doc/nrf/libraries/modem/location.rst

```diff
@@ -0,0 +1,198 @@
+* :kconfig:`CONFIG_LOCATION_METHOD_CELLULAR` - Enables cellular location method.
+* :kconfig:`CONFIG_LOCATION_METHOD_WIFI` - Enables WiFi location method.
+* :kconfig:`CONFIG_WIFI` - Enable WiFi for Zephyr.
+
+The following options control the use of GNSS assistance data:
+
+* :kconfig:`CONFIG_LOCATION_METHOD_GNSS_AGPS_EXTERNAL` - Enables A-GPS data retrieval from an external source which the application implements separately.
+* :kconfig:`CONFIG_NRF_CLOUD_AGPS` - Enables A-GPS data retrieval from `nRF Cloud`_.
+* :kconfig:`CONFIG_NRF_CLOUD_PGPS` - Enables P-GPS data retrieval from `nRF Cloud`_.
+
+The following options control the transport used with `nRF Cloud`_:
+
+* :kconfig:`CONFIG_NRF_CLOUD_REST` - Uses REST APIs to communicate with `nRF Cloud`_.
+* :kconfig:`CONFIG_NRF_CLOUD_MQTT` - Uses MQTT transport to communicate with `nRF Cloud`_.
+* :kconfig:`CONFIG_REST_CLIENT` - Enable :ref:`lib_rest_client` library.
+
+Both cellular and WiFi location services are selected utilizing runtime configuration but the available services must be configured first.
+For cellular location services, use at least one of the following sets of options and configure corresponding authentication parameters (more details and configuration options can be found from :ref:`lib_multicell_location`):
+
+* :kconfig:`CONFIG_MULTICELL_LOCATION_SERVICE_NRF_CLOUD`
+* :kconfig:`CONFIG_MULTICELL_LOCATION_SERVICE_HERE` and :kconfig:`CONFIG_MULTICELL_LOCATION_HERE_API_KEY` (see below other authentication options)
+* :kconfig:`CONFIG_MULTICELL_LOCATION_SERVICE_SKYHOOK` and :kconfig:`CONFIG_MULTICELL_LOCATION_SKYHOOK_API_KEY`
+* :kconfig:`CONFIG_MULTICELL_LOCATION_SERVICE_POLTE` and :kconfig:`CONFIG_MULTICELL_LOCATION_POLTE_CUSTOMER_ID` and :kconfig:`CONFIG_MULTICELL_LOCATION_POLTE_API_TOKEN`
+
+For WiFi location services, use at least one of the following sets of options and configure corresponding authentication parameters:
+
+* :kconfig:`CONFIG_LOCATION_METHOD_WIFI_SERVICE_NRF_CLOUD`
+* :kconfig:`CONFIG_LOCATION_METHOD_WIFI_SERVICE_HERE` and :kconfig:`CONFIG_LOCATION_METHOD_WIFI_SERVICE_HERE_API_KEY`
+* :kconfig:`CONFIG_LOCATION_METHOD_WIFI_SERVICE_SKYHOOK` and :kconfig:`CONFIG_LOCATION_METHOD_WIFI_SERVICE_SKYHOOK_API_KEY`
+
+Following WiFi service related options can usually have default values:
+
+* :kconfig:`CONFIG_LOCATION_METHOD_WIFI_SERVICE_HERE_HOSTNAME`
+* :kconfig:`CONFIG_LOCATION_METHOD_WIFI_SERVICE_HERE_TLS_SEC_TAG`
+* :kconfig:`CONFIG_LOCATION_METHOD_WIFI_SERVICE_SKYHOOK_HOSTNAME`
+* :kconfig:`CONFIG_LOCATION_METHOD_WIFI_SERVICE_SKYHOOK_TLS_SEC_TAG`
+
+Usage
+*****
+
+Using Location library is rather easy. First you need to initialize the library with :c:func:`location_init`.
+
+Secondly, you need to set configuration (:c:struct:`loc_config`) including location method configurations (:c:struct:`loc_method_config`).
+This is achieved easily by setting first default values by using :c:func:`loc_config_defaults_set` and c:func:`loc_config_method_defaults_set`,
```

Comment:
```suggestion
This is easily achieved by setting first default values by using :c:func:`loc_config_defaults_set` and c:func:`loc_config_method_defaults_set`,
```

Reply:
Not relevant after Pekka's modification which I'll apply